test/Microsoft.AspNetCore.Razor.Test/Parser/Html/HtmlDocumentTest.cs

```diff
@@ -341,6 +343,85 @@ public void ParseDocumentDoesNotRenderExtraNewlineAtTheEndTextTagInVerbatimBlock
+                new MarkupBlock(
+                    Factory.EmptyHtml(),
+                    new StatementBlock(
+                        Factory.CodeTransition(),
+                        Factory.MetaCode("{").Accepts(AcceptedCharacters.None),
+                        new MarkupBlock(
+                            new MarkupTagBlock(
+                                Factory.MarkupTransition("<text>")),
+                            Factory.Markup("Blah").Accepts(AcceptedCharacters.None),
+                            new MarkupTagBlock(
+                                Factory.MarkupTransition("</text>")),
+                            Factory.Markup("\r\n").Accepts(AcceptedCharacters.None)),
+                        new MarkupBlock(
+                            new MarkupTagBlock(
+                                Factory.Markup("<input/>").Accepts(AcceptedCharacters.None)),
+                            Factory.Markup("\r\n").Accepts(AcceptedCharacters.None)),
+                        Factory.EmptyCSharp().AsStatement(),
+                        Factory.MetaCode("}").Accepts(AcceptedCharacters.None)),
+                    BlockFactory.MarkupTagBlock("<html>")));
+        }
+
+        [Fact]
+        public void ParseDocumentRendersExtraNewlineAtTheEndTextTagInVerbatimBlockIfFollowedByMarkupTransition()
+        {
+            ParseDocumentTest("@{<text>Blah</text>\r\n@: Bleh\r\n}<html>",
+                new MarkupBlock(
+                    Factory.EmptyHtml(),
+                    new StatementBlock(
+                        Factory.CodeTransition(),
+                        Factory.MetaCode("{").Accepts(AcceptedCharacters.None),
+                        new MarkupBlock(
+                            new MarkupTagBlock(
+                                Factory.MarkupTransition("<text>")),
+                            Factory.Markup("Blah").Accepts(AcceptedCharacters.None),
+                            new MarkupTagBlock(
+                                Factory.MarkupTransition("</text>")),
+                            Factory.Markup("\r\n").Accepts(AcceptedCharacters.None)),
+                        new MarkupBlock(
+                            Factory.MarkupTransition(),
+                            Factory.MetaMarkup(":", HtmlSymbolType.Colon),
+                            Factory.Markup(" Bleh\r\n")
+                                .With(new SingleLineMarkupEditHandler(CSharpLanguageCharacteristics.Instance.TokenizeString))
+                                .Accepts(AcceptedCharacters.None)),
+                        Factory.EmptyCSharp().AsStatement(),
+                        Factory.MetaCode("}").Accepts(AcceptedCharacters.None)),
+                    BlockFactory.MarkupTagBlock("<html>")));
+        }
+
+        [Fact]
+        public void ParseDocumentRendersNewlineAtTheEndHtmlTagInACSharpBlock()
+        {
+            ParseDocumentTest("@using(var foo = new Foo()){<div>Blah</div>\r\n<div>Bleh</div>\r\n}<html>",
```

Comment:
Both these cases are already covered by `ParseDocumentDoesNotRenderExtraNewlineAtTheEndTextTagInVerbatimBlockIfFollowedByCSharp` and `ParseDocumentRendersExtraNewlineAtTheEndTextTagInVerbatimBlockIfFollowedByHtml`. I added this case intentionally for the normal html tag case. I'll check if we have an existing test for this case and remove this.